lib/features/posts/data/firestorage/posts_firestorage.dart

```diff
@@ -0,0 +1,23 @@
+import 'dart:io';
+
+import 'package:firebase_storage/firebase_storage.dart';
+import 'package:instreal/features/posts/data/firestorage/firestorage_event.dart';
+
+abstract interface class PostsFirestorage {
+  Stream<FirestorageEvent> upload(File image);
+}
+
+class PostsFirestorageImpl implements PostsFirestorage {
+  PostsFirestorageImpl({FirebaseStorage? storage})
+      : storageRef = (storage ?? FirebaseStorage.instance).ref("posts");
+
+  final Reference storageRef;
+
+  @override
+  Stream<FirestorageEvent> upload(File image) {
```

Comment:
This needs to be implemented properly. I'm thinking of creating a type for the argument that contains metadata + filepath or byte array of the file.

@aliyazdi75 @billyandco What do you guys think?

Reply:
@aliyazdi75

Reply:
Yes I think keeping metadata is useful and byte array.